langchain/document_loaders/blob_transformers/youtube_to_audio.py

```diff
@@ -0,0 +1,33 @@
+from langchain.document_loaders.base import BaseBlobTransformer
+from langchain.document_loaders.blob_loaders import Blob
+
+
+class YoutubeToAudioTransformer(BaseBlobTransformer):
```

Comment:
VideoToAudioTransformer

Reply:
one subtle point here: `yt_dlp` takes a YouTube url specifically. 

so the functionality more specific than `VideoToAudioTransformer`.

the flow here is something like:
1/ we can add a `WebUrlBlobLoader` (as you note that we should add [here](https://www.notion.so/LangChain-Documents-786e0b2762694ca786889556ea4586ba#1de4c8a6cedf4aaab8d501e32afe4b81).
2/ we then extract the `url` from the blob and pass it here

Reply:
Makes sense! 

We could either use a specific Blob Loader or a more generic WebURLBlobLoader that takes in a download handler. 

The former could be simpler as a first pass so we don't have to think about how to implement a generic download handler etc

```python
class YoutubeAudioLoader(BlobLoader)?
  def __init__(urls):
    ...

  def yield_blobs() -> Iterable[Blob]: # <-- Yields Blobs that have data attribute populated with the audio
   pass
```